util/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,99 @@
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	agentpayload "github.com/DataDog/agent-payload/gogen"
+	agentkubernetes "github.com/DataDog/datadog-agent/pkg/metadata/kubernetes"
+	agentkubelet "github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-process-agent/util/cache"
+)
+
+const (
+	cacheKey                   = "kubernetes_meta"
+	kubernetesServiceTagPrefix = "kube_service:"
+	kubernetesMetaTTL          = 3 * time.Minute
+)
+
+var (
+	// ErrKubernetesNotAvailable if the machine is not running in Kubernetes.
+	ErrKubernetesNotAvailable = errors.New("kubernetes not available")
+
+	globalKubeUtil *agentkubelet.KubeUtil
+	lastKubeErr    string
+)
+
+// InitKubeUtil initializes a global kubeUtil used by later function calls.
+// We keep track of our own global kubeUtil even though the agentkubelet already does to prevent a noisy log
+func InitKubeUtil() error {
+	if ku, err := agentkubelet.GetKubeUtil(); err == nil {
+		globalKubeUtil = ku
```

Comment:
can we debug log the error here for support purposes (we log at INFO by default, so it won't show)